presto-main/src/main/java/io/prestosql/event/QueryMonitor.java

```diff
@@ -309,15 +325,34 @@ private QueryContext createQueryContext(SessionRepresentation session, Optional<
 
     private static QueryIOMetadata getQueryIOMetadata(QueryInfo queryInfo)
     {
+        Multimap<FragmentNode, OperatorStats> tableOperatorStats = extractStatsForTableScan(queryInfo);
```

Comment:
since we're now keying on `FragmentNode`, we could exclude the "table scan" logic here, and just invoke an `extractStats` method that treats all nodes equally. I think that may be easier to reason about. 

The visitor can just have 

```
                    protected Void visitPlan(PlanNode node, Collection<OperatorStats> parentStats)
                    {
                        if (parentStats != null) {
                            tableScanOperatorStats.putAll(new FragmentNode(fragment.getId(), node.getId()), allOperatorStats.getOrDefault(node.getId(), parentStats));
                        }

                        for (PlanNode child : node.getSources()) {
                            child.accept(this, allOperatorStats.getOrDefault(node.getId(), parentStats));
                        }
                        return null;
                    }
``` 

But if you'd like to make this method stand-out specifically for table scan stats usecase (and save some bytes), the current implementation seems fine too.

Reply:
I like this style better.  I updated to code like the above.